spring-integration-ip/src/main/java/org/springframework/integration/ip/udp/UnicastSendingMessageHandler.java

```diff
@@ -277,7 +296,9 @@ public void startAckThread() {
 
 	protected void send(DatagramPacket packet) throws Exception {
 		DatagramSocket socket = this.getSocket();
-		packet.setSocketAddress(this.getDestinationAddress());
+		if (this.socketExpression == null) {
+			packet.setSocketAddress(this.getDestinationAddress());
```

Comment:
@artembilan Do you want me to refactor this in whole `udp` package or only in places I've changed?